test/FastTests/Utils/TimeParsing.cs

```diff
@@ -151,7 +154,130 @@ public void TimeOnly(string date, int hh, int mm, int ss, int ms)
+                    DateTimeStyles.RoundtripKind);
+
+                var bytes = Encoding.UTF8.GetBytes(tested);
+                fixed (byte* buffer = bytes)
+                {
+                    Assert.Equal(LazyStringParser.Result.DateTime,
+                        LazyStringParser.TryParseDateTime(buffer, bytes.Length, out DateTime time, out _, properlyParseThreeDigitsMilliseconds: true));
+                    Assert.Equal(expectedAfterFormatting, time);
+                }
+            }
+        }
+
+        [RavenTheory(RavenTestCategory.Core)]
+        [InlineDataWithRandomSeed]
+        public void CanParseValidRandomDate(int seed)
+        {
+            var r = new Random(seed);
+
+            var dt = GetRandomDate(r);
+
+            var formatsToRead = new Dictionary<string, DateTimeKind>
+            {
+                {DefaultFormat.DateTimeFormatsToRead[0], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[1], DateTimeKind.Unspecified},
+                {DefaultFormat.DateTimeFormatsToRead[2], DateTimeKind.Local},
+                {DefaultFormat.DateTimeFormatsToRead[3], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[4], DateTimeKind.Unspecified},
+                {DefaultFormat.DateTimeFormatsToRead[5], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[6], DateTimeKind.Utc},
+            };
+            
+            Assert.Equal(formatsToRead.Count, DefaultFormat.DateTimeFormatsToRead.Length);
+
+            foreach (var dateTimeFormat in formatsToRead)
+            {
+                string tested = dt.ToString(dateTimeFormat.Key);
+
+                var bytes = Encoding.UTF8.GetBytes(tested);
+                fixed (byte* buffer = bytes)
+                {
+                    var parseResult = LazyStringParser.TryParseDateTime(buffer, bytes.Length, out var dateTime, out var dateTimeOffset, properlyParseThreeDigitsMilliseconds: true);
+                    
+                    Assert.NotEqual(LazyStringParser.Result.Failed, parseResult);
+
+                    switch (parseResult)
+                    {
+                        case LazyStringParser.Result.DateTime:
+
+                            Assert.Equal(tested, dateTime.ToString(dateTimeFormat.Key));
+                            break;
+                        case LazyStringParser.Result.DateTimeOffset:
+
+                            Assert.Equal(tested, dateTimeOffset.ToString(dateTimeFormat.Key));
+                            break;
+                    }
+                }
+            }
+        }
+
+        private static DateTime GetRandomDate(Random random, int minYear = 1900, int maxYear = 2099)
+        {
+            var year = random.Next(minYear, maxYear);
+            var month = random.Next(1, 12);
+            var noOfDaysInMonth = DateTime.DaysInMonth(year, month);
+            var day = random.Next(1, noOfDaysInMonth);
+
+            return new DateTime(year, month, day);
```

Comment:
shouldn't we put milliseconds here? if not then it is set to 0:00.00 then?

Reply:
Yes. Fixed